tools/python/find_untranslated_strings.py

```diff
@@ -0,0 +1,302 @@
+        self.translations_by_language = defaultdict(dict) # dict<lang, dict<key, translation>>
+        self.comments_and_tags = defaultdict(dict)
+        self.with_english = []
+        self.all_langs = set()
+        self.duplicates = {} # dict<lang, TransAndKey>
+        self.keys_in_order = []
+        self._read_file()
+        self._populate_translations_by_langs()
+        self._find_duplicates()
+        self.most_duplicated = []
+        self._find_most_duplicated()
+        self.similarity_indices = []
+        self._find_most_similar()
+
+
+    def _read_file(self):
+        with open(self.strings_path) as strings:
+            for line in strings:
+                line = line.strip()
+                if not line:
+                    continue
+                if line.startswith("[["):
+                    self.keys_in_order.append(line)
+                    continue
+                if line.startswith("["):
+                    if line in self.translations:
+                        print("Duplicate key {}".format(line))
+                        continue
+                    self.translations[line] = {}
+                    current_key = line
+                    self.keys_in_order.append(current_key)
+
+                if TRANSLATION.match(line):
+                    lang, tran = self._lang_and_translation(line)
+                    self.translations[current_key][lang] = tran
+
+                    self.all_langs.add(lang)
+                    if line.startswith("en = "):
+                        self.with_english.append(current_key)
+                    continue
+
+                if line.startswith("comment") or line.startswith("tags"):
+                    lang, value = self._lang_and_translation(line)
+                    self.comments_and_tags[current_key][lang] = value
+                    continue
+
+
+    def print_statistics(self):
+        stats = map(lambda x: (x, len(self.translations[x])), self.translations.keys())
+        stats.sort(key=lambda x: x[1], reverse=True)
+
+        for k, v in stats:
+            print("{0}\t{1}".format(k, v))
+
+
+    def print_duplicates(self):
+        print(self._header("Duplicates:"))
+        for lang, trans_and_keys in self.duplicates.items():
+            print("{0}\n {1}\n".format("=" * (len(lang) + 2), lang))
+            last_one = ""
+            keys = []
+            for tr in trans_and_keys:
+                if last_one != tr.translation:
+                    self._print_keys_for_duplicates(keys, last_one)
+                    keys = []
+                last_one = tr.translation
+                keys.append(tr.key)
+            self._print_keys_for_duplicates(keys, last_one)
+
+
+    def _print_keys_for_duplicates(self, keys, last_one):
+        if last_one:
+            print("{0}: {1}\n".format(", ".join(keys), last_one))
+
+
+    def _process_string(self, string):
+        if MANY_DOTS.search(string):
```

Comment:
Why crash the script in this case? It is quite possible that someone might intentionally put more than three full stops into a string.